drivers: uart: Add low power uart module (#2591)
nrfconnect/sdk-nrf: samples/peripheral/lpuart/README.rst

.. _lpuart_sample:

Low Power UART sample
#####################

Sample demonstrates capabilities of low power UART module (:ref:`uart_nrf_lpuart`).

Overview
********

> Sample implements simple loopback using single UART instance.Sample by default

Author: done

has console and logging disabled to show low power consumption while having UART
active.

Requirements
************

* One of the following development boards:

  * |nRF52840DK|
  * |nRF9160DK|

> * Pins shortend

Author: done

.. _lpuart_sample:

Low Power UART sample
#####################

The Low Power UART sample demonstrates the capabilities of the :ref:`uart_nrf_lpuart` module.

Overview
********

The sample implements a simple loopback using a single UART instance.
The sample has console and logging disabled by default, to demonstrate low power consumption while having UART active.

Requirements
************

* One of the following development boards:

>   * |nRF52840DK|

Reviewer: See this on how to add supported dev kits -> https://github.com/nrfconnect/sdk-nrf/pull/2632/files#diff-954121af6470d736bd2a2ef278913f88 (line 31)

Reviewer 2: fixed

  * |nRF9160DK|

* The following pins shorted:

  * TX (Arduino Digital Pin 10) with RX (Arduino Digital Pin 11)
  * Request Pin (Arduino Digital Pin 12) with Response Pin (Arduino Digital Pin 13)

Building and running
********************
.. |sample path| replace:: :file:`samples/peripheral/lpuart`

.. include:: /includes/build_and_run.txt

Testing
=======

> After programming the sample to your board, test it by performing the following steps:

Reviewer: ```suggestion
After programming the sample to your development kit, test it by performing the following steps:
```
We are not using 'board' anymore, use 'development kit'.

Reviewer 2: fixed

> 1. Connect the Logic Analyzer to the shorted pins, to confirm UART activity.

Reviewer: ```suggestion
1. Connect a logic analyzer to the shorted pins, to confirm UART activity.
```
Probably no reason to capitalize this.
Also, logic analyzer should be probably mentioned in the requirements section for this sample.

Reviewer 2: fixed

> 2. Measure the current, to confirm that the power consumption indicates that high frequency clock is disabled during the idle stage.

Reviewer: ```suggestion
2. Measure the current to confirm that the power consumption indicates that high frequency clock is disabled during the idle stage.
```

Reviewer 2: fixed

   During the idle stage, the UART receiver is ready to start reception, as the request pin wakes it up.

Dependencies
************

> This sample is using following |NCS| subsystems:

Reviewer: ```suggestion
This sample uses the following |NCS| subsystem:
```
I'd rather just call it 'driver' to be honest.

Reviewer 2: fixed

.. _lpuart_sample:

Low Power UART
##############

The Low Power UART sample demonstrates the capabilities of the :ref:`uart_nrf_lpuart` module.

Overview
********

The sample implements a simple loopback using a single UART instance.
The sample has console and logging disabled by default, to demonstrate low power consumption while having UART active.

Requirements
************

* One of the following development boards:

.. include:: /includes/boardname_tables/sample_boardnames.txt
   :start-after: set18_start

>    :end-before: set18_end

Reviewer 3: Shouldn't these two be `set22_start` and `set22_end`, respectively?

Author: well spotted. fixed

> Low Power UART sample

Reviewer: Remove ''Sample'' from the title

Reviewer 2: fixed
